pkg/types/reference.go

```diff
@@ -0,0 +1,100 @@
+package types
+
+import (
+	"fmt"
+	"go/token"
+	"go/types"
+	"reflect"
+	"strings"
+
+	"github.com/crossplane-contrib/terrajet/pkg/terraform/resource"
+
+	"github.com/pkg/errors"
+	"golang.org/x/tools/go/packages"
+
+	"github.com/crossplane-contrib/terrajet/pkg/comments"
+)
+
+const (
+	// PackagePathXPCommonAPIs is the go path for the Crossplane Runtime package
+	// with common APIs
+	PackagePathXPCommonAPIs = "github.com/crossplane/crossplane-runtime/apis/common/v1"
+
+	loadMode = packages.NeedName | packages.NeedImports | packages.NeedTypes
+)
+
+var typeXPRef types.Type
+var typeXPSelector types.Type
+var commentOptional *comments.Comment
+
+func (g *Builder) getReferenceFields(t *types.TypeName, f *types.Var, r resource.FieldReferenceConfiguration) ([]*types.Var, []string) {
```

Comment:
I think we can start writing unit tests for utilities we don't mean to revisit. Could you add some for this function?

Reply:
nitpick

```suggestion
func (g *Builder) GenerateReferenceFields(t *types.TypeName, f *types.Var, r resource.FieldReferenceConfiguration) ([]*types.Var, []string) {
```

Reply:
Do you want to make this public, even only used in the same package? Why?